pkg/dns/client/proxy.go

```diff
@@ -45,24 +48,33 @@ func newDNSProxy(protocol string, resolver *LocalDNSServer) (*dnsProxy, error) {
 		protocol: protocol,
 		resolver: resolver,
 	}
+	addr := resolver.addr
+	if addr == "" {
+		addr = "localhost:15053"
+	}
+	if strings.HasSuffix(addr, ":53") && os.Getuid() != 0 {
+		log.Error("DNS address :53 and not running as root, use default")
+		addr = "localhost:15053"
+	}
+	p.addr = addr
 
 	var err error
 	p.serveMux.Handle(".", p)
 	p.server.Handler = p.serveMux
 	if protocol == "udp" {
-		p.server.PacketConn, err = net.ListenPacket("udp", "localhost:15053")
+		p.server.PacketConn, err = net.ListenPacket("udp", addr)
 	} else {
-		p.server.Listener, err = net.Listen("tcp", "localhost:15053")
+		p.server.Listener, err = net.Listen("tcp", addr)
 	}
 	if err != nil {
-		log.Errorf("Failed to listen on %s port 15053: %v", protocol, err)
+		log.Errorf("Failed to listen on %s port %s: %v", protocol, addr, err)
```

Comment:
```suggestion
		log.Errorf("Failed to listen on %s at %s: %v", protocol, addr, err)
```
very nit: its not just port now